src/Layouts/Partials/FillItem.vala

```diff
@@ -0,0 +1,332 @@
+    attach (visible_button, 1, 0, 1, 1);
+    attach (delete_button, 2, 0, 1, 1);
+
+    get_style_context ().add_class ("fill-chooser-cont");
+  }
+
+  private void create_event_bindings () {
+    delete_button.clicked.connect (on_delete_item);
+    visible_button.clicked.connect (toggle_visibility);
+
+    blending_mode_popover_items.row_activated.connect (on_row_activated);
+    blending_mode_popover_items.row_selected.connect (on_popover_item_selected);
+
+    selected_blending_mode_cont.clicked.connect (() => { on_show_popover ("blending_mode"); });
+
+    selected_color.clicked.connect (() => { on_show_popover ("color"); });
+    color_chooser_widget.notify["rgba"].connect (on_color_changed);
+
+    current_opacity_cont.clicked.connect (() => { on_show_popover ("opacity"); });
+    opacity_slider.value_changed.connect (on_opacity_changed);
+
+    show_options_button.clicked.connect (() => { on_show_popover ("blending_mode"); });
+
+    model.notify.connect (on_model_changed);
+  }
+
+  private void on_color_changed () {
+    var selectedColor = color_chooser_widget.rgba;
+
+    color = "#%02X%02X%02X".printf (
+      (int) (selectedColor.red * 255),
+      (int) (selectedColor.green * 255),
+      (int) (selectedColor.blue * 255)
+    );
+  }
+
+  private void on_model_changed () {
+    model.list_model.update_fills ();
+  }
+
+  private void on_row_activated (Gtk.ListBoxRow? item) {
+    var fillItem = (Akira.Layouts.Partials.BlendingModeItem) item.get_child ();
+    blending_mode = fillItem.mode;
+    popover.hide ();
+  }
+
+  private void on_popover_item_selected (Gtk.ListBoxRow? item) {
+  }
+
+  private void on_show_popover (string target) {
+    foreach (Gtk.Widget elem in popover.get_children ()) {
+      popover.remove (elem);
+    }
+
+    switch (target) {
+      case "blending_mode":
+        popover.width_request = get_allocated_width ();
+        popover.relative_to = selected_blending_mode_cont;
+        popover.child = blending_mode_popover_items;
+        break;
+
+      case "opacity":
+        popover.width_request = get_allocated_width ();
+        popover.relative_to = current_opacity_cont;
+        popover.child = opacity_slider;
+        break;
+
+      case "color":
+        popover.width_request = get_allocated_width ();
+        popover.relative_to = selected_color;
+        popover.child = color_picker;
+        break;
+    }
+
+    if (!popover.visible) {
+      popover.show_all ();
+    } else {
+      popover.hide ();
+    }
+  }
+
+  private void on_delete_item () {
+    remove_item (model);
+  }
+
+  private void toggle_visibility () {
+    visible = !visible;
+  }
+
+  private void on_opacity_changed (Gtk.Range slider) {
+    opacity = (uint) slider.get_value ();
+  }
+
+  private void set_selected_color_background () {
+    try {
+      var provider = new Gtk.CssProvider ();
+      var context = selected_color.get_style_context ();
+
+      var css = """
+        .selected-color {
```

Comment:
Any reason for hardcoding the css here?

Reply:
this is the non deprecated way to do it with gtk. Totally annoying, yes.

Reply:
I don't know whether this is the *only* way to achieve this, but I just copied it from @Philip-Scott's Spice Up, since there was a similar use case in there! I found that the GTK+ docs online was lacking regarding this topic ("changing a button's background") and so I ended up taking inspiration from what was done by @Philip-Scott. That's the power of OSS :)